src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    } else if (this.length === 1) {
+      return String(this[0]);
+    }
+
+    let result;
+    const separatorToString = String(separator);
```

Comment:
Do you really need to convert it to string?

Reply:
No need. Because even if it is a number, we still concatenate it with a string. Forgot to delete.